sentry-android-core/src/main/java/io/sentry/android/core/internal/measurement/cpu/CpuMeasurementCollector.java

```diff
@@ -0,0 +1,125 @@
+package io.sentry.android.core.internal.measurement.cpu;
+
+import android.os.Build;
+import android.os.Process;
+import android.os.SystemClock;
+import android.system.Os;
+import android.system.OsConstants;
+import io.sentry.ITransaction;
+import io.sentry.SentryLevel;
+import io.sentry.SentryOptions;
+import io.sentry.measurement.BackgroundAwareMeasurementCollector;
+import io.sentry.measurement.MeasurementBackgroundService;
+import io.sentry.measurement.MeasurementBackgroundServiceType;
+import io.sentry.measurement.MeasurementContext;
+import io.sentry.protocol.MeasurementValue;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@ApiStatus.Internal
+public final class CpuMeasurementCollector extends BackgroundAwareMeasurementCollector {
+
+  private final SentryOptions options;
+  private List<MeasurementBackgroundServiceType> listenToTypes;
+  private @Nullable Long startRealtimeNanos;
+  private @Nullable Long startElapsedTimeMs;
+
+  public CpuMeasurementCollector(
+      @NotNull SentryOptions options, @NotNull MeasurementBackgroundService backgroundService) {
+    super(backgroundService);
+    this.options = options;
+    this.listenToTypes = Arrays.asList(MeasurementBackgroundServiceType.CPU);
+  }
+
+  @Override
+  protected List<MeasurementBackgroundServiceType> listenToTypes() {
+    return listenToTypes;
+  }
+
+  @Override
+  protected void onTransactionStartedInternal(@NotNull ITransaction transaction) {
+    // TODO 8 - 23 μs
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
+      startRealtimeNanos = SystemClock.elapsedRealtimeNanos();
+      startElapsedTimeMs = Process.getElapsedCpuTime();
+    }
+  }
+
+  @Override
+  protected @Nullable Map<String, MeasurementValue> onTransactionFinishedInternal(
+      @NotNull ITransaction transaction, @NotNull MeasurementContext context) {
+    HashMap<String, MeasurementValue> results = new HashMap<>();
+
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1
+        && startRealtimeNanos != null
+        && startElapsedTimeMs != null) {
+      // TODO SystemClock.elapsedRealtimeNanos() + Process.getElapsedCpuTime() take 9 - 15 μs
+      Long diffRealtimeNanos = SystemClock.elapsedRealtimeNanos() - startRealtimeNanos;
```

Comment:
This seems like our best bet as it's very fast and can be done synchronously. Probably also less risk of being taken away than reading `/proc/self/stat`

NOTE: The difference for `Process.getElapsedCpuTime()` might be higher than for `SystemClock.elapsedRealtimeNanos()` if multiple cores are under load.